WooCommerce/Classes/ViewRelated/Dashboard/Free Trial/FreeTrialSurvey/FreeTrialSurveyViewModel.swift

```diff
@@ -0,0 +1,79 @@
+import SwiftUI
+
+/// View model for `FreeTrialSurveyView`
+///
+final class FreeTrialSurveyViewModel: ObservableObject {
+    @Published private(set) var selectedAnswer: SurveyAnswer?
+    @Published var otherReasonSpecified: String = ""
+
+    private let analytics: Analytics
+    private let source: FreeTrialSurveyCoordinator.Source
+
+    init(source: FreeTrialSurveyCoordinator.Source,
+         analytics: Analytics = ServiceLocator.analytics) {
+        self.source = source
+        self.analytics = analytics
+    }
+
+    var answers: [SurveyAnswer] {
+        SurveyAnswer.allCases
+    }
+
+    var feedbackSelected: Bool {
+        otherReasonSpecified.isNotEmpty || selectedAnswer != nil
```

Comment:
We may want to update this check to ensure that a reason is specified if the selected answer is "Other":

```suggestion
        guard selectedAnswer != nil else {
            return false
        }
        if selectedAnswer == .otherReasons {
            return otherReasonSpecified.isNotEmpty
        }
        return true
```

Reply:
Good point. I will include this in a future PR with unit tests.